===========================
How to use 0-add_integer.py
===========================

This module tests the fanctionality of ''add_integer(a, b=98)'' function.

Usage
=====

''add_integer(...)'' takes 2 integers and returns their sum. It is equivalent to using
the ''+'' operator.

For Numbers(integers)
=====================

::

    >>> add_integer = __import__('0-add_integer').add_integer
    >>> add_integer(2, 8)
    10

::

    >>> add_integer(-2, 8)
    6

For floating numbers.
=====================

Floats should be casted before their addition.

::

    >>> add_integer(2.5, 8.5)
    10

::
    >>> add_integer(-2.5, 8.5)
    6

For Non-numbers
===============

''add_integer()'' only takes up floats or integers. A TypeError is raised if
a non-integer or non-float is entered.

::

    >>> add_integer('num', 8)
    Traceback (most recent call last):
    TypeError: a must be an integer

::

    >>> add_integer(2, 'num')
    Traceback (most recent call last):
    TypeError: b must be an integer

For NaN numbers
===============

A ValueError is raised when a NaN number is passes to the function.

::

    >>> add_integer(float('nan'))
    Traceback (most recent call last):
    ValueError: cannot convert float NaN to integer

::

    >>> add_integer(-2, float('nan'))
    Traceback (most recent call last):
    ValueError: cannot convert float NaN to integer

For Infinity values
===================

An OverflowError is raised when infinity numbers are entered.

::

    >>> add_integer(float('inf'))
    Traceback (most recent call last):
    OverflowError: cannot convert float infinity to integer

::

    >>> add_integer(-2, float('inf'))
    Traceback (most recent call last):
    OverflowError: cannot convert float infinity to integer

No arguments given
==================

Raises a TypeError when no arguments are provided or when a is missing.

::

    >>> add_integer(None)
    Traceback (most recent call last):
    TypeError: a must be an integer

::

    >>> add_integer(2)
    100

::

    >>> add_integer(None, 8)
    Traceback (most recent call last):
    TypeError: a must be an integer
